edx_notifications/lib/publisher.py

```diff
@@ -6,32 +6,32 @@
 any new methods here will also be exposed to xBlocks!!!!
 """
 
-import logging
+from __future__ import absolute_import
+
+import copy
 import types
+import logging
 import datetime
-import pytz
-import copy
-from contracts import contract
 
-from django.db.models.query import QuerySet
+import pytz
 
-from edx_notifications.channels.channel import get_notification_channel
+from contracts import contract
 from edx_notifications import const
-from edx_notifications.stores.store import notification_store
+from django.db.models.query import QuerySet
+from edx_notifications.data import NotificationType, NotificationMessage, NotificationCallbackTimer
+from edx_notifications.scopes import resolve_user_scope, has_user_scope_resolver
 from edx_notifications.exceptions import ItemNotFoundError
+from edx_notifications.stores.store import notification_store
+from edx_notifications.channels.channel import get_notification_channel
+from edx_notifications.renderers.renderer import register_renderer
 
-from edx_notifications.data import (
-    NotificationType,
-    NotificationMessage,
-    NotificationCallbackTimer,
-)
+log = logging.getLogger(__name__)
 
-from edx_notifications.renderers.renderer import (
-    register_renderer
-)
-from edx_notifications.scopes import resolve_user_scope, has_user_scope_resolver
 
-log = logging.getLogger(__name__)
+try:
+    basestring  # pylint: disable=invalid-name
+except NameError:
+    basestring = str  # pylint: disable=redefined-builtin, invalid-name
```

Comment:
Can we replace this try catch block with `basestring = six.string_types` ?

Reply:
`six.string_types` is a tuple, so it doesn't work.

Reply:
```python
if six.PY3:
    basestring = str
```
This could work.

Reply:
```
if six.PY3:
    basestring = str
```
Then please use this as it seems better than try block